components/usage/pkg/apiv1/usage-report.go

```diff
@@ -44,11 +47,21 @@ type ReportGenerator struct {
 	nowFunc func() time.Time
 }
 
-func (g *ReportGenerator) GenerateUsageReport(ctx context.Context, from, to time.Time) (UsageReport, error) {
+func (g *ReportGenerator) GenerateUsageReport(ctx context.Context, from, to time.Time) (*UsageReport, error) {
 	now := g.nowFunc().UTC()
-	log.Infof("Gathering usage data from %s to %s", from, to)
 
-	report := UsageReport{
+	// Sanity check: from <= now
+	if now.Before(from) {
+		return nil, status.Errorf(codes.InvalidArgument, "Now must be before from")
```

Comment:
Error message is wrong:

```suggestion
		return nil, status.Errorf(codes.InvalidArgument, "Now must be after from")
```